File: 2022/4/a.ts
const input = Deno.readTextFileSync("input-a.txt");
const lines = input.split("\n");

const assignments = [];

for (const line of lines) {
  const [left, right] = line.split(",");
  assignments.push({ left, right });
}

let overlaps = 0;

for (const assignment of assignments) {
  const [leftStart, leftEnd] = assignment.left.split("-").map(Number);
  const [rightStart, rightEnd] = assignment.right.split("-").map(Number);

  const overlap = (leftStart >= rightStart && leftEnd <= rightEnd) || (rightStart >= leftStart && rightEnd <= leftEnd);

  if (overlap) overlaps++;
}

console.log(overlaps);

File: 2022/4/a.py
input = open('2022/4/input-a.txt', 'r')
lines = input.readlines()

score = 0
for line in lines:
    left, right = line.split(',')

    leftStart, leftEnd = map(int, left.split('-'))
    rightStart, rightEnd = map(int, right.split('-'))

    if (leftStart >= rightStart and leftEnd <= rightEnd) or (rightStart >= leftStart and rightEnd <= leftEnd):
        score += 1

print(score)
